diesel/src/sqlite/connection/mod.rs

```diff
@@ -67,17 +67,19 @@ impl Connection for SqliteConnection {
         Self::Backend: HasSqlType<T::SqlType>,
         U: Queryable<T::SqlType, Self::Backend>,
     {
-        let statement = try!(self.prepare_query(&source.as_query()));
-        let mut statement_ref = statement.borrow_mut();
-        StatementIterator::new(&mut statement_ref).collect()
+        let mut statement = try!(self.prepare_query(&source.as_query()));
+        let statement_use = StatementUse::new(&mut statement);
+        let iter = StatementIterator::new(statement_use);
+        iter.collect()
```

Comment:
Yeah, it claims that `statement` is dropped with a life reference. Which is wrong, borrowck should know is wrong (even without non-lexical lifetimes), and I think is probably a bug in rustc. FWIW this also compiles:

```rust
let x = StatementIterator::new(statement_use).collect();
x
```

But clippy doesn't like that one.